app/views/layouts/application.html.erb

```diff
@@ -10,10 +10,21 @@
 
   <body>
     <nav class = "topnav">
+      <%= link_to "Treasure Trove", "/" %>
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register" %>
+
+      <% if current_user %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Profile", "/profile" %>
+        <%= link_to "Logout", "/logout" %>
+        <div>Logged in as <%= current_user.name %></div>
+      <% else %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Register", "/register" %>
+        <%= link_to "Login", "/login" %>
+      <% end %>
```

Comment:
I assume the Cart link is duplicated because one of the other user types won't be able to access it so you're getting it ready?

Reply:
@DanielEFrampton Yep. A registered default user (ie. current_user) and visitor (ie. non-logged-in user via the 'else') should be able to see the cart, but not all roles will have a cart (ie. a merchant or admin).

Reply:
👍

Reply:
Sorry...Just saw this. ignore my comment